src/test/benchmarks/io/vertx/benchmarks/CombinerExecutorBenchmark.java

```diff
@@ -39,39 +46,49 @@
 public class CombinerExecutorBenchmark extends BenchmarkBase {
 
   private Executor<Object> exec;
-  private Executor.Action<Object> action;
 
   private CountDownLatch latch = new CountDownLatch(1);
 
+  @Param({ "0", "1", "10" })
+  private int actionWork;
+
   @Setup
   public void setup() throws Exception {
     exec = new CombinerExecutor<>(new Object());
-    action = state -> {
-      Blackhole.consumeCPU(0);
-      return null;
-    };
-    CountDownLatch l = new CountDownLatch(2);
-    new Thread(() -> {
-      exec.submit(state -> {
-        l.countDown();
-        try {
-          latch.await();
-        } catch (InterruptedException e) {
-          e.printStackTrace();
-        }
-        return null;
-      });
-    }).start();
-    l.await(20, TimeUnit.SECONDS);
   }
 
   @TearDown
   public void tearDown() {
     latch.countDown();
   }
 
+  @AuxCounters
+  @State(Scope.Thread)
+  public static class ActionStatistics {
+
+    public long submitted;
+    public long completed;
+
+    private Executor.Action<Object> action;
+
+    @Setup
+    public void clean(CombinerExecutorBenchmark benchmark, Blackhole bh) {
+      submitted = completed = 0;
```

Comment:
Sadly `completed` cannot be a plain field, because will be read by the submitter thread (that's not by any mean the executor!!!) - but given that actions are accessed in serialized fashion I can just use setOpaque/lazySet in order to keep it lightweight

Reply:
actually having it global is better so it applies consistently to all actions and we don't need to deal with different policiies submitted by different threads